modules/exploits/multi/http/drupal_drupageddon.rb

```diff
@@ -33,33 +48,134 @@ def initialize(info={})
       'Privileged'     => false,
       'Platform'       => ['php'],
       'Arch'           => ARCH_PHP,
-      'Targets'        => [['Drupal 7.0 - 7.31',{}]],
+      'Targets'        =>
+        [
+          ['Drupal 7.0 - 7.31 (form-cache PHP injection method)', {}],
+          ['Drupal 7.0 - 7.31 (user-post PHP injection method)', {}]
+        ],
       'DisclosureDate' => 'Oct 15 2014',
       'DefaultTarget'  => 0
     ))
 
     register_options(
-    [
-      OptString.new('TARGETURI', [ true, "The target URI of the Drupal installation", '/'])
-    ])
+      [
+        OptString.new('TARGETURI', [ true, "The target URI of the Drupal installation", '/'])
+      ])
 
     register_advanced_options(
-    [
-      OptString.new('ADMIN_ROLE', [ true, "The administrator role", 'administrator']),
-      OptInt.new('ITER', [ true, "Hash iterations (2^ITER)", 10])
-    ])
+      [
+        OptInt.new('Wait', [true, "Number of seconds to wait before triggering the payload sent (form-cache method only).", 5]),
+        OptString.new('AdminRole', [ true, "The administrator role (user-post method only)", 'administrator']),
+        OptInt.new('Iter', [ true, "Hash iterations (2^ITER, user-post method only))", 10])
+      ])
+  end
+
+  ##
+  # Form-cache PHP injection method
+  # See details on:
+  # https://www.whitewinterwolf.com/posts/2017/11/16/drupageddon-revisited-a-new-path-from-sql-injection-to-remote-command-execution-cve-2014-3704/
```

Comment:
can we add this link to the references of the module so it will be visible when using info?

Reply:
Yes I can. I hesitated whether the references section was only about official and high-level advisories, so in doubt I kept this more technical link for the source-code.

Two other things I am not sure about:

- The two methods use different advanced options, I don't know if there is a way to show only the options matching the currently selected target (for now I simply updated the description).

- In the `info` output, there is no carriage return in the Description field which is rendered as a single block of text, making it not very readable. The other modules I checked seem to have the same problem so I'm not sure if there is anything I can do to improve this.